### PR TITLE
docs: use "driver" instead of "provider" in design.md

### PR DESCRIPTION
**Pull request by vangent**
Updates #2548.

### CODE REVIEW

File: internal/docs/design.md [hunk 1]
```diff
@@ -115,12 +115,12 @@ it is needed.
 
 The exception we permit is URL scheme registration as documented under
 [URLs](#urls). The amount of boilerplate setup code required for URL muxes for
-multiple providers without use of a tool like Wire is an unreasonable burden for
+multiple drovers without use of a tool like Wire is an unreasonable burden for
```
Comment (polinasok) [inline internal/docs/design.md]: Is this a type or a compromise solution drover = providers + driver? :)

File: internal/docs/design.md [hunk 2]
```diff
@@ -142,15 +142,15 @@ The naming convention is `<provider><product><api>`, where:
...
+    *   Encouraged when it helps distinguish the package from the service's own
         package name (e.g., `s3blob` not `s3`).
 
 ## Portable Type Constructors
 
-Portable type constructors are the functions defined in provider-specific
-packages that end users call to get an instance of the portable type. For
-example, `gcsblob.OpenBucket`, which returns an instance of the `*blob.Bucket`
-portable type backed by GCS.
+Portable type constructors are the functions defined in driver packages that end
```
Comment (polinasok) [inline internal/docs/design.md]: service-specific driver packages?

File: internal/docs/design.md [hunk 3]
```diff
@@ -490,26 +490,26 @@ We ended up going with `As` because:
...
 The Go CDK APIs will end up exposing functionality that is not supported by all
-provider implementations. In addition, some functionality details will differ
-across providers. Some theoretical examples using [`blob.Bucket`][]:
+services. In addition, some functionality details will differ across services.
+Some theoretical examples using [`blob.Bucket`][]:
 
-1.  **Top-level APIs**: There might be a provider implementation that supports
-    reads, but not writes or deletes.
-1.  **Data fields**. Some providers may support key/value metadata associated
+1.  **Top-level APIs**: There might be a services that supports reads, but not
```
Comment (polinasok) [inline internal/docs/design.md]: service (singular)

File: internal/docs/design.md [hunk 4]
```diff
@@ -742,12 +741,11 @@ wiki page. We also adopt the following guidelines:
...
-Since our goal is for users to be able to use provider implementations
-interchangeably, it is critical that they behave similarly. To this end, each
-portable API (e.g., `blob`) must provide a suite of conformance tests that
-provider implementations should run. The conformance tests should be
-comprehensive; provider implementations should not need additional unit tests
-for the core driver semantics.
+Since our goal is for users to be able to use drivers interchangeably, it is
+critical that they behave similarly. To this end, each portable API (e.g.,
+`blob`) must provide a suite of conformance tests that driver implementations
+should run. The conformance tests should be comprehensive; drvers should not
```
Comment (polinasok) [inline internal/docs/design.md]: type: drvers

File: internal/docs/design.md (hunk 1 above)
Comment (vangent) [reply internal/docs/design.md]: Ha - no. Fixed to `driver`.

File: internal/docs/design.md (hunk 2 above)
Comment (vangent) [reply internal/docs/design.md]: I think driver packages is fine here. It's shorter, and some drivers are not necessarily service-specific (for example, the Mongo driver is usable against multiple services).

File: internal/docs/design.md (hunk 3 above)
Comment (vangent) [reply internal/docs/design.md]: Done.

File: internal/docs/design.md (hunk 4 above)
Comment (vangent) [reply internal/docs/design.md]: Done.

**Pull request merged**